### PR TITLE
Update types to work with typescript 4.8+

### LINKED ISSUE #877: Error buliding with TypeScript 4.8
Hi there! We're getting the following error building the latest version (6.0.5) of the core package:

```
Error: node_modules/@dnd-kit/core/dist/sensors/types.d.ts(56,20): error TS2344: Type 'T' does not satisfy the constraint 'Object'.
```

It looks like one of the generic T parameters is missing an `extends Object` or `extends {}`, which TS 4.8 is stricter about.

### PR DESCRIPTION
**Pull request by kilbiller**
Fix #877

### CODE REVIEW

File: packages/utilities/src/hooks/useLatestValue.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import {useIsomorphicLayoutEffect} from './useIsomorphicLayoutEffect';
 
 export function useLatestValue<T extends any>(
   value: T,
-  dependencies = [value]
+  dependencies: unknown[] = [value]
```
Comment (kilbiller) [inline packages/utilities/src/hooks/useLatestValue.ts:7]: Not sure about this, I tried to be as strict as possible.

**Comment by SimonSchick** [comment]
This seems sensible, can this be merged & published? Kinda blocked on this 👀

**Comment by Asifm95** [comment]
Same here as well. Any ETA on when this will be merged?

**Comment by osovv** [comment]
Is there any news on this?

**Comment by re-thc** [comment]
https://github.com/clauderic/dnd-kit/pull/948 this has been merged for TS 4.8 support. Can we close this?

**Comment by kilbiller** [comment]
Sure

**Pull request closed without merging**